types/htmlbars-inline-precompile/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for htmlbars-inline-precompile 1.0
+// Project: ember-cli-htmlbars-inline-precompile
+// Definitions by: Chris Krycho <https://github.com/chriskrycho>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// This is a bit of a funky one: it's from a [Babel plugin], but is exported for
+// Ember applications as the module `"htmlbars-inline-precompile"`. It acts
+// like a tagged string from the perspective of consumers, but is actually an
+// AST transformation which generates a function as its output. That function in
+// turn [generates a string or array of strings][output] to use with the Ember
+// testing helper `this.render()`.
+//
+// [Babel plugin]: https://github.com/ember-cli/babel-plugin-htmlbars-inline-precompile#babel-plugin-htmlbars-inline-precompile-
+// [output]: https://github.com/emberjs/ember-test-helpers/blob/77f9a53da9d8c19a85b3122788caadbcc59274c2/lib/ember-test-helpers/-legacy-overrides.js#L17-L42
+
+export default function hbs(tagged: TemplateStringsArray): string | string[];
```

Comment:
Consumers write an import statement like normal as if it were an ES6-style `export default`:

```
import hbs from 'htmlbars-inline-precompile';
```

Then at build time, it's transformed (literally "precompiled"). But from the perspective of the consumers, it's just a default export tagged template string function.